src/kbmod/search/raw_image.cpp

```diff
@@ -1,235 +1,184 @@
-    if (fits_open_file(&fptr, layerPath.c_str(), READONLY, &status)) {
-        fits_report_error(stderr, status);
-        throw std::runtime_error("Could not open FITS file to read RawImage");
-    }
-
-    // Read image dimensions.
-    long dimensions[2];
-    if (fits_read_keys_lng(fptr, "NAXIS", 1, 2, dimensions, &file_not_found, &status))
-        fits_report_error(stderr, status);
-    width = dimensions[0];
-    height = dimensions[1];
-
-    // Read in the image.
-    pixels = std::vector<float>(width * height);
-    if (fits_read_img(fptr, TFLOAT, 1, get_npixels(), &nullval, pixels.data(), &anynull, &status))
-        fits_report_error(stderr, status);
-
-    load_time_from_file(fptr);
-    if (fits_close_file(fptr, &status)) fits_report_error(stderr, status);
-
-    // If we are reading from a sublayer and did not find a time, try the overall header.
-    if (obstime < 0.0) {
-        if (fits_open_file(&fptr, file_path.c_str(), READONLY, &status))
-            throw std::runtime_error("Could not open FITS file to read RawImage");
-        load_time_from_file(fptr);
-        if (fits_close_file(fptr, &status)) fits_report_error(stderr, status);
-    }
+    if (sumWeights == 0.0) return NO_DATA;
+    return total / sumWeights;
 }
 
-void RawImage::save_to_file(const std::string& filename) {
-    fitsfile* fptr;
-    int status = 0;
-    long naxes[2] = {0, 0};
-
-    fits_create_file(&fptr, filename.c_str(), &status);
+RawImage RawImage::create_stamp(const Point& p, const int radius, const bool interpolate,
+                                const bool keep_no_data) const {
+    if (radius < 0) throw std::runtime_error("stamp radius must be at least 0");
 
-    // If we are unable to create the file, check if it already exists
-    // and, if so, delete it and retry the create.
-    if (status == 105) {
-        status = 0;
-        fits_open_file(&fptr, filename.c_str(), READWRITE, &status);
-        if (status == 0) {
-            fits_delete_file(fptr, &status);
-            fits_create_file(&fptr, filename.c_str(), &status);
+    const int dim = radius * 2 + 1;
+    // can't address this instance of non-uniform index handling with Point
+    // and Index, because at a base level it adopts a different definition of
+    // the pixel grid to coordinate system transformation.
+    auto [corner, anchor, h, w] = indexing::anchored_block({(int)p.y, (int)p.x}, radius, width, height);
+
+    Image stamp;
+    if (keep_no_data)
+        stamp = Image::Constant(dim, dim, NO_DATA);
+    else
+        stamp = Image::Zero(dim, dim);
+    stamp.block(anchor.j, anchor.i, h, w) = image.block(corner.j, corner.i, h, w);
+
+    if (interpolate) {
+        for (int yoff = 0; yoff < dim; ++yoff) {
+            for (int xoff = 0; xoff < dim; ++xoff) {
+                // I think the {} create a temporary, but I don't know how bad that is
+                // would it be the same if we had interpolate just take 2 floats?
```

Comment:
As discussed in the meeting, I think we can get rid of interpolated stamps in a later PR.